test/worker.js

```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+const { expect } = require('chai')
+const metrics = require('..')
+
+metrics.start()
+
+const sab = new SharedArrayBuffer(16)
+const typedArray = new Int32Array(sab)
+
+Atomics.wait(typedArray, 0, 0, 100)
```

Comment:
What's going on here? It looks like you're waiting on a SAB that hasn't been shared with any other thread.

Reply:
Is it possible to wait without a `SharedArrayBuffer`? When I try I get this error: `TypeError: [object Int32Array] is not a shared typed array.`.

Reply:
no, @bengl this is causing a sleep for 100ms with the last argument

Reply:
Got it.

Reply:
But ... Why do this rather than a setTimeout, especially since we have a setImmediate next?

Reply:
`setTimeout` doesn't block the event loop. We hook into `prepare` and `check` so that we're able to know if it's actually blocked or just idle.